src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Trunc.cs

```diff
@@ -141,12 +153,22 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
                 if (type0.IsTable)
                 {
                     // Ensure we have a one-column table of numerics
-                    fValid &= CheckNumericColumnType(type0, args[0], errors, ref nodeToCoercedTypeMap);
-
-                    returnType = context.Features.HasFlag(Features.ConsistentOneColumnTableResult)
-                        ? DType.CreateTable(new TypedName(DType.Number, new DName(ColumnName_ValueStr)))
-                        : type0;
-                }
+                    fValid &= CheckNumericColumnType(type0, args[0], errors, ref nodeToCoercedTypeMap);
+
+                    if (nodeToCoercedTypeMap?.Any() ?? false)
+                    {
+                        // Now set the coerced type to a table with numeric column type with the same name as in the argument.
+                        returnType = nodeToCoercedTypeMap[args[0]];
+                    }
+                    else
+                    {
+                        returnType = type0;
+                    }
+

```

Comment:
this pattern is repeated a few places, should it be moved to a helper?

Reply:
This is the subject of another PR, https://github.com/microsoft/Power-Fx/pull/1256.  I think I would do the cleanup there or in still another PR, depending on how we want to manage change.